react-components/src/hooks/useCameraNavigation.tsx

```diff
@@ -0,0 +1,68 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import { type CogniteCadModel } from '@cognite/reveal';
+import { useReveal } from '../components/RevealContainer/RevealContext';
+import { useFdmSdk } from '../components/RevealContainer/SDKProvider';
+import { SYSTEM_3D_EDGE_SOURCE, type InModel3dEdgeProperties } from '../utilities/globalDataModels';
+
+export type CameraNavigationActions = {
+  fitCameraToAllModels: () => void;
+  fitCameraToModelNode: (revisionId: number, nodeId: number) => Promise<void>;
+  fitCameraToInstance: (externalId: string, space: string) => Promise<void>;
+};
+
+export const useCameraNavigation = (): CameraNavigationActions => {
+  const viewer = useReveal();
+  const fdmSDK = useFdmSdk();
+
+  const fitCameraToAllModels = (): void => {
+    const models = viewer.models;
+    if (models.length === 0) {
+      return;
+    }
+    viewer.fitCameraToModels(models, undefined, true);
+  };
+
+  const fitCameraToModelNode = async (revisionId: number, nodeId: number): Promise<void> => {
+    const model = viewer.models.find((m) => m.revisionId === revisionId);
```

Comment:
Can we support cases where multiple instances of one model is loaded? This will only pick the first of the models if there are multiple.